src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -521,11 +523,31 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 local_files_only=local_files_only,
                 use_auth_token=use_auth_token,
                 revision=revision,
+                user_agent=user_agent,
             )
+            _commit_hash = config_dict.pop("_commit_hash", None)
 
             # retrieve all folder_names that contain relevant files
             folder_names = [k for k, v in config_dict.items() if isinstance(v, list)]
 
+            # verify that every model folder of the pipeline is present
+            pipeline_is_cached = True
+            for k, v in config_dict.items():
+                component_is_expected = isinstance(v, list) and v[0] is not None
+                component_is_None_passed = k in kwargs and kwargs.get(k) is None
+                if component_is_expected and not component_is_None_passed:
+                    pipeline_is_cached = (
+                        try_to_load_from_cache(
+                            pretrained_model_name_or_path, cache_dir=cache_dir, subfolder=k, revision=_commit_hash
+                        )
+                        is not None
+                    )
+            # TODO(Patrick) - need to check here that every subfolder is compatible
+            # There can still be edge cases with `variant` and `safetensors`
```

Comment:
I don't think this is enough when variants come into play. The cached folder may exist but the variant files don't. Not sure how to deal with this. We can expand to the individual files we need, but in order to do that we need to retrieve the list of files from the repo. Or we could retrieve the tree from the local cache folder and verify that we have the set we need.